zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -177,14 +185,21 @@ def working_set(
     links=(),
     index=None,
     allow_hosts=('*',),
+    cache_storage=_global_working_set_cache,
 ):
     """Helper function to build a working set.
 
     Return an instance of `pkg_resources.WorkingSet`.
 
-    Results are cached in the `_working_set_cache` dict at module level.
+    Results are cached in the mapping provided in the `cache_storage` argument.
+    This defaults to a global dict at this module. If `None` is passed then
+    caching is disabled.
```

Comment:
Is there a reason to even have the option of a module global dict cache? That seems like a foot-gun to me.